queue.go

```diff
@@ -231,18 +233,15 @@ func (queue *redisQueue) StartConsuming(prefetchLimit int, pollDuration time.Dur
 	return true
 }
 
-func (queue *redisQueue) StopConsuming() bool {
-	if queue.deliveryChan == nil || queue.consumingStopped {
-		return false // not consuming or already stopped
-	}
-
+func (queue *redisQueue) StopConsuming() *sync.WaitGroup {
 	queue.consumingStopped = true
-	return true
+	return &queue.stopWg
```

Comment:
This waitgroup must be a private thing only. It's better to return a `<-chan struct{}` (similar to how it works in `context.Done()`), so a user won't have an ability to shoot into her own's foot, by calling `WaitGroup.Done / WaitGroup.Add` herself.